C compilation error: "undefined symbol" for module public const

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->

<details>
<summary>V version: V 0.4.12 d9f1e33, press to see full `v doctor` output</summary>

|V full version      |V 0.4.12 caf7d413ad1a68aafc20e2a2786d08c30a3964d7.d9f1e33
|:-------------------|:-------------------
|OS                  |linux, Linux version 6.17.9-300.fc43.x86_64 (mockbuild@928180b471f54176bbf3623ccb67b438) (gcc (GCC) 15.2.1 20251111 (Red Hat 15.2.1-4), GNU ld version 2.45.1-1.fc43) #1 SMP PREEMPT_DYNAMIC Mon Nov 24 23:31:27 UTC 2025
|Processor           |16 cpus, 64bit, little endian, AMD Ryzen 7 7800X3D 8-Core Processor
|Memory              |22.23GB/30.94GB
|                    |
|V executable        |/home/yann/Apps/v/v
|V last modified time|2025-12-03 17:07:33
|                    |
|V home dir          |OK, value: /home/yann/Apps/v
|VMODULES            |OK, value: /home/yann/.vmodules
|VTMP                |OK, value: /tmp/v_1000
|Current working dir |OK, value: /home/yann/devel/perso/v/bug_cgen_pub_const_mod
|                    |
|Git version         |git version 2.52.0
|V git status        |weekly.2025.43-241-gd9f1e336
|.git/config present |true
|                    |
|cc version          |cc (GCC) 15.2.1 20251111 (Red Hat 15.2.1-4)
|gcc version         |gcc (GCC) 15.2.1 20251111 (Red Hat 15.2.1-4)
|clang version       |clang version 21.1.6 (Fedora 21.1.6-1.fc43)
|tcc version         |tcc version 0.9.28rc 2025-02-13 HEAD@f8bd136d (x86_64 Linux)
|tcc git status      |thirdparty-linux-amd64 696c1d84
|emcc version        |N/A
|glibc version       |ldd (GNU libc) 2.42

</details>

**What did you do?**
`./v -g -o vdbg cmd/v && ./vdbg  main.v && main`
```v
module main

import my_utils

fn main() {
	println('Hello World!')

	v := f64(1.0/8)
	println('${my_utils.my_round(v, 2)}')
}

```

with `modules/my_utils/my_utils.v`
```v
module my_utils

import strconv

pub const my_round = strconv.f64_to_str_lnd1
```

**What did you see?**
```
================== C compilation error (from tcc): ==============
cc: /tmp/v_1000/main.01KBJM3FGZMHEFYQWXTRAN0QE0.tmp.c:4960: warning: cast between pointer and integer of different size
cc: tcc: error: undefined symbol '_const_my_utils__my_round'
...
cc: tcc: error: undefined symbol '_const_my_utils__my_round'
(note: the original output was 4 lines long; it was truncated to its first 2 lines + the last line)
=================================================================
Try passing `-g` when compiling, to see a .v file:line information, that correlates more with the C error.
(Alternatively, pass `-show-c-output`, to print the full C error message).
builder error: 
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
```

**What did you expect to see?**

A successful compilation

**Additional notes**

Interestingly part of my project compiles with the reference to the const in the module but fails on a given V source file.
Hopefully the example to reproduce the issue is very succint.

> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/markused/walker.v
+++ b/vlib/v/markused/walker.v
@@ -454,7 +454,9 @@ fn (mut w Walker) expr(node_ ast.Expr) {
 		}
 		ast.CallExpr {
 			w.call_expr(mut node)
-			if node.name == 'json.decode' {
+			if node.is_fn_a_const {
+				w.mark_const_as_used(node.name)
+			} else if node.name == 'json.decode' {
 				w.mark_by_type((node.args[0].expr as ast.TypeNode).typ)
 			} else if node.name == 'json.encode' && node.args[0].typ != 0 {
 				sym := w.table.final_sym(node.args[0].typ)

--- a/vlib/v/tests/consts/const_markused_test.v
+++ b/vlib/v/tests/consts/const_markused_test.v
@@ -1,0 +1,6 @@
+import my_utils
+
+fn test_main() {
+	v := f64(1.0 / 8)
+	assert my_utils.my_round(v, 2) == '0.13'
+}

--- a/vlib/v/tests/consts/my_utils/my_utils.v
+++ b/vlib/v/tests/consts/my_utils/my_utils.v
@@ -1,0 +1,5 @@
+module my_utils
+
+import strconv
+
+pub const my_round = strconv.f64_to_str_lnd1

